subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/InHouseInstrumentationTestRunner.kt

```diff
@@ -268,13 +268,6 @@ abstract class InHouseInstrumentationTestRunner :
             }
 
             onWaiterRetry = { }
-
-            /**
-             * Большинство ексепшенов от espresso оборачивается в UITestFrameworkException
-             */
-            waiterAllowedExceptions = waiterAllowedExceptions.plus(
-                UITestFrameworkException::class.java
-            )
```

Comment:
We lost any unknown error in retry and did one more time.

"unknown" = else case in `ReportFriendlyFailureHandler.handle`

Example:

- test: `click()` 

It runs in retry loop
- activity: 
  - handles click
  - startActivity() -> fail ! (This is the original error)
- Espresso failure handler (`ReportFriendlyFailureHandler`) wraps any unknown error to `UITestFrameworkException`
- Retry loop skips it as allowed

We go into the second loop, but we in another state on the screen. Original error is lost. Happy debugging!

Reply:
@elya-a, it was the cause of the lost error